File: go/trafficlight/machine/mom.go
package machine

//	"fmt"

type MOM struct {
	m TrafficLight
}

func NewMOM() *MOM {
	mom := &MOM{}
	mom.m = New()
	return mom
}

//func (mom *MOM) Start(w http.ResponseWriter, r *http.Request) {
func (mom *MOM) Start() {
	mom.m.Start(mom)
}

func (mom *MOM) Stop() {
	mom.m.Stop()
}

File: javascript/trafficlight/trafficlight/mom.js
const TrafficLightController = require("./machine");

class Mom {
    constructor() {
        this.mom = new TrafficLightController()
    }

    start() {
        this.mom.start(this.mom)
    }

    stop() {
        this.mom.stop()
    }
}

module.exports = Mom